# TUMWIZERE RICHARD MAY21/MIT/589U_DEATHS DUE TO COVID-19 PANDEMIC IN AMERICAS REGION 
Descriptive Statistics 


In [1]:
import pandas as pd
import numpy as np

In [2]:
df=pd.read_csv("WHO COVID-19 global table data July 9th 2021 at 9.50.54 AM.csv")

In [3]:
df

Name       WHO Region  Cases - cumulative total  \
0                      Global              NaN                 184820132   
1    United States of America         Americas                  33429396   
2                       India  South-East Asia                  30709557   
3                      Brazil         Americas                  18855015   
4          Russian Federation           Europe                   5707452   
..                        ...              ...                       ...   
233              Saint Helena           Africa                         0   
234                   Tokelau  Western Pacific                         0   
235                     Tonga  Western Pacific                         0   
236              Turkmenistan           Europe                         0   
237                    Tuvalu  Western Pacific                         0   

     Cases - cumulative total per 100000 population  \
0                                       2371.148567   
1                                      10099.430000   
2                                       2225.320000   
3                                       8870.470000   
4                                       3910.970000   
..                                              ...   
233                                        0.000000   
234                                        0.000000   
235                                        0.000000   
236                                        0.000000   
237                                        0.000000   

     Cases - newly reported in last 7 days  \
0                                  2857227   
1                                    96394   
2                                   297923   
3                                   341710   
4                                   169310   
..                                     ...   
233                                      0   
234                                      0   
235                                      0   
236                                      0   
237                                      0   

     Cases - newly reported in last 7 days per 100000 population  \
0                                            36.656773             
1                                            29.120000             
2                                            21.590000             
3                                           160.760000             
4                                           116.020000             
..                                                 ...             
233                                           0.000000             
234                                           0.000000             
235                                           0.000000             
236                                           0.000000             
237                                           0.000000             

     Cases - newly reported in last 24 hours  Deaths - cumulative total  \
0                                     484439                    4002209   
1                                      36990                     600930   
2                                      45892                     405028   
3                                      62504                     526892   
4                                      24818                     140775   
..                                       ...                        ...   
233                                        0                          0   
234                                        0                          0   
235                                        0                          0   
236                                        0                          0   
237                                        0                          0   

     Deaths - cumulative total per 100000 population  \
0                                          51.346312   
1                                         181.550000   
2           

In [4]:
richard = df.loc[(df['WHO Region'] == 'Americas')]
richard

Name WHO Region  Cases - cumulative total  \
1              United States of America   Americas                  33429396   
3                                Brazil   Americas                  18855015   
8                             Argentina   Americas                   4574340   
9                              Colombia   Americas                   4402582   
15                               Mexico   Americas                   2549862   
19                                 Peru   Americas                   2069051   
22                                Chile   Americas                   1576336   
24                               Canada   Americas                   1418080   
46                              Ecuador   Americas                    465029   
48     Bolivia (Plurinational State of)   Americas                    448213   
49                             Paraguay   Americas                    431367   
53                               Panama   Americas                    410004   
55                           Costa Rica   Americas                    375812   
56                              Uruguay   Americas                    374019   
62                   Dominican Republic   Americas                    330941   
63                            Guatemala   Americas                    308273   
68   Venezuela (Bolivarian Republic of)   Americas                    279813   
72                             Honduras   Americas                    270020   
79                                 Cuba   Americas                    214577   
89                          Puerto Rico   Americas                    140412   
108                         El Salvador   Americas                     80235   
115                             Jamaica   Americas                     50456   
124                 Trinidad and Tobago   Americas                     33805   
129                       French Guiana   Americas                     27996   
134                            Suriname   Americas                     22581   
136                              Guyana   Americas                     20568   
139                               Haiti   Americas                     19172   
141                          Guadeloupe   Americas                     17684   
150                              Belize   Americas                     13409   
151                             Bahamas   Americas                     12965   
153                          Martinique   Americas                     12396   
154                             Curaçao   Americas                     12350   
157                               Aruba   Americas                     11144   
166                           Nicaragua   Americas                      6819   
173                         Saint Lucia   Americas                      5363   
178                            Barbados   Americas                      4108   
181        United States Virgin Islands   Americas                      3942   
184                        Sint Maarten   Americas                      2630   
186                             Bermuda   Americas                      2521   
187            Turks and Caicos Islands   Americas                      2425   
188                        Saint Martin   Americas                      2420   
193    Saint Vincent and the Grenadines   Americas                      2240   
196                             Bonaire   Americas                      1626   
197                 Antigua and Barbuda   Americas                      1265   
198                    Saint Barthélemy   Americas                      1046   
201              British Virgin Islands   Americas                       793   
203                      Cayman Islands   Americas                       615   
204               Saint Kitts and Nevis   Americas                       517   
208                            Dominica   Americas                       197   
210                             Gre

In [5]:
richard.head()

Name WHO Region  Cases - cumulative total  \
1   United States of America   Americas                  33429396   
3                     Brazil   Americas                  18855015   
8                  Argentina   Americas                   4574340   
9                   Colombia   Americas                   4402582   
15                    Mexico   Americas                   2549862   

    Cases - cumulative total per 100000 population  \
1                                         10099.43   
3                                          8870.47   
8                                         10121.17   
9                                          8652.38   
15                                         1977.67   

    Cases - newly reported in last 7 days  \
1                                   96394   
3                                  341710   
8                                  126639   
9                                  189508   
15                                  36698   

    Cases - newly reported in last 7 days per 100000 population  \
1                                               29.12             
3                                              160.76             
8                                              280.20             
9                                              372.44             
15                                              28.46             

    Cases - newly reported in last 24 hours  Deaths - cumulative total  \
1                                     36990                     600930   
3                                     62504                     526892   
8                                     21590                      96983   
9                                     26721                     110019   
15                                     7989                     233958   

    Deaths - cumulative total per 100000 population  \
1                                            181.55   
3                                            247.88   
8                                            214.58   
9                                            216.22   
15                                           181.46   

    Deaths - newly reported in last 7 days  \
1                                     1518   
3                                    10907   
8                                     3315   
9                                     4085   
15                                    1155   

    Deaths - newly reported in last 7 days per 100000 population  \
1                                                0.46              
3                                                5.13              
8                                                7.33              
9                                                8.03              
15                                               0.90              

    Deaths - newly reported in last 24 hours Transmission Classification  
1                                        473      Community transmission  
3                                       1780      Community transmission  
8                                        462      Community transmission  
9                                        553      Community transmission  
15                                       269      Community transmission

In [6]:
richard.tail()

Name WHO Region  Cases - cumulative total  \
213  Falkland Islands (Malvinas)   Americas                        60   
216    Saint Pierre and Miquelon   Americas                        26   
217                   Montserrat   Americas                        20   
218               Sint Eustatius   Americas                        20   
220                         Saba   Americas                         7   

     Cases - cumulative total per 100000 population  \
213                                         1722.65   
216                                          448.66   
217                                          400.08   
218                                          637.15   
220                                          362.13   

     Cases - newly reported in last 7 days  \
213                                      0   
216                                      0   
217                                      0   
218                                      0   
220                                      0   

     Cases - newly reported in last 7 days per 100000 population  \
213                                                0.0             
216                                                0.0             
217                                                0.0             
218                                                0.0             
220                                                0.0             

     Cases - newly reported in last 24 hours  Deaths - cumulative total  \
213                                        0                          0   
216                                        0                          0   
217                                        0                          1   
218                                        0                          0   
220                                        0                          0   

     Deaths - cumulative total per 100000 population  \
213                                              0.0   
216                                              0.0   
217                                             20.0   
218                                              0.0   
220                                              0.0   

     Deaths - newly reported in last 7 days  \
213                                       0   
216                                       0   
217                                       0   
218                                       0   
220                                       0   

     Deaths - newly reported in last 7 days per 100000 population  \
213                                                0.0              
216                                                0.0              
217                                                0.0              
218                                                0.0              
220                                                0.0              

     Deaths - newly reported in last 24 hours Transmission Classification  
213                                         0                    No cases  
216                                         0              Sporadic cases  
217                                         0                    No cases  
218                                         0                    No cases  
220                                         0                    No cases

In [7]:
richard.dtypes

Name                                                             object
WHO Region                                                       object
Cases - cumulative total                                          int64
Cases - cumulative total per 100000 population                  float64
Cases - newly reported in last 7 days                             int64
Cases - newly reported in last 7 days per 100000 population     float64
Cases - newly reported in last 24 hours                           int64
Deaths - cumulative total                                         int64
Deaths - cumulative total per 100000 population                 float64
Deaths - newly reported in last 7 days                            int64
Deaths - newly reported in last 7 days per 100000 population    float64
Deaths - newly reported in last 24 hours                          int64
Transmission Classification                                      object
dtype: object

In [8]:
richard.describe()

Cases - cumulative total  \
count              5.600000e+01   
mean               1.308872e+06   
std                5.106454e+06   
min                7.000000e+00   
25%                2.086500e+03   
50%                1.554650e+04   
75%                3.417105e+05   
max                3.342940e+07   

       Cases - cumulative total per 100000 population  \
count                                       56.000000   
mean                                      4089.820893   
std                                       3304.720199   
min                                        102.940000   
25%                                       1395.055000   
50%                                       3173.840000   
75%                                       6266.230000   
max                                      10767.070000   

       Cases - newly reported in last 7 days  \
count                              56.000000   
mean                            16953.767857   
std                             54830.205019   
min                                 0.000000   
25%                                 8.500000   
50%                               299.000000   
75%                              7555.000000   
max                            341710.000000   

       Cases - newly reported in last 7 days per 100000 population  \
count                                          56.000000             
mean                                           92.195000             
std                                           218.058399             
min                                             0.000000             
25%                                             8.752500             
50%                                            33.295000             
75%                                            95.160000             
max                                          1587.460000             

       Cases - newly reported in last 24 hours  Deaths - cumulative total  \
count                                56.000000                  56.000000   
mean                               3189.839286               34411.107143   
std                               10440.116521              111939.915985   
min                                   0.000000                   0.000000   
25%                                   0.000000                  17.750000   
50%                                   4.500000                 261.000000   
75%                                1095.500000                5949.000000   
max                               62504.000000              600930.000000   

       Deaths - cumulative total per 100000 population  \
count                                        56.000000   
mean                                         77.013750   
std                                          95.449194   
min                                           0.000000   
25%                                          11.202500   
50%                                          53.210000   
75%                                          93.312500   
max                                         587.130000   

       Deaths - newly reported in last 7 days  \
count                               56.000000   
mean                               460.053571   
std                               1604.636197   
min                                  0.000000   
25%                                  0.000000   
50%                                  3.000000   
75%                                121.000000   
max                              10907.000000   

       Deaths - newly reported in last 7 days per 100000 population  \
count                                          56.000000              
mean                                            1.581071              
std                                             2.417386              
min                                             0.000000              
25%                                             0.000000              
50%                         

In [9]:
richard.shape

(56, 13)

In [10]:
richard['WHO Region'].value_counts().to_frame()

WHO Region
Americas          56

In [11]:
richard['Transmission Classification'].value_counts().to_frame()

Transmission Classification
Community transmission                           42
Sporadic cases                                    5
Clusters of cases                                 5
No cases                                          4

In [12]:
subsetdata_set = richard [['Name', 'Cases - cumulative total','Deaths - cumulative total']]
subsetdata_set.head()

Name  Cases - cumulative total  \
1   United States of America                  33429396   
3                     Brazil                  18855015   
8                  Argentina                   4574340   
9                   Colombia                   4402582   
15                    Mexico                   2549862   

    Deaths - cumulative total  
1                      600930  
3                      526892  
8                       96983  
9                      110019  
15                     233958

In [13]:
subsetdata_set = richard [['Name', 'Cases - newly reported in last 24 hours', 'Transmission Classification']]
subsetdata_set.head()

Name  Cases - newly reported in last 24 hours  \
1   United States of America                                    36990   
3                     Brazil                                    62504   
8                  Argentina                                    21590   
9                   Colombia                                    26721   
15                    Mexico                                     7989   

   Transmission Classification  
1       Community transmission  
3       Community transmission  
8       Community transmission  
9       Community transmission  
15      Community transmission

In [14]:
sorted_data = richard [['Name', 'Cases - newly reported in last 24 hours', 'Transmission Classification']]
subsetdata_set.head()

Name  Cases - newly reported in last 24 hours  \
1   United States of America                                    36990   
3                     Brazil                                    62504   
8                  Argentina                                    21590   
9                   Colombia                                    26721   
15                    Mexico                                     7989   

   Transmission Classification  
1       Community transmission  
3       Community transmission  
8       Community transmission  
9       Community transmission  
15      Community transmission

# Simple Linear Regression

y=mx+c 

y= independent_variable , m= slope , x= independent_variable , c=y-intercept

In [15]:
from sklearn.linear_model import LinearRegression

In [16]:
#Using identified cases to predict deaths in the next 24 hours
x = richard[['Cases - newly reported in last 24 hours']] 
y = richard['Deaths - newly reported in last 24 hours'] 
lm = LinearRegression()

lm.fit(x,y)

Yhat=lm.predict(x)
Yhat[0:5]


array([ 882.80199219, 1493.97196106,  513.90579714,  636.81530212,
        188.10338487])

In [17]:
# Calculationg Coeficient
lm.coef_

array([0.0239543])

In [18]:
#Intercept
lm.intercept_

-3.2675048918436147

In [19]:
#calculatin R-Squared
lm.score(x,y)

0.9196431842409537

In [20]:
#Calculating R-squared
print('The R-squared is: ', lm.score(x, y))

The R-squared is:  0.9196431842409537


In [46]:
from sklearn.metrics import mean_squared_error #for calculating MSE
from math import sqrt 


In [22]:
#calculating MSE
y=lm.predict(x)
MSE = mean_squared_error (richard['Deaths - newly reported in last 24 hours'], y)
print('MSE is', MSE)


MSE is 5367.296984035462


In [47]:
#Calculating Root mean squared error
print('Root Mean Squared Error:',sqrt(MSE))

Root Mean Squared Error: 73.26183852481087


# Multiple Linear Regression

In [23]:
z = richard[['Cases - newly reported in last 7 days' , 'Cases - cumulative total per 100000 population' , 'Cases - newly reported in last 24 hours']]

In [35]:
lm.fit(z, richard['Deaths - newly reported in last 24 hours'])

LinearRegression()

In [36]:
# Calculationg Coeficient
lm.coef_

array([ 0.00317971, -0.00283341,  0.0082509 ])

In [37]:
#Intercept
lm.intercept_

4.503845358040024

In [38]:
#calculatin R-Squared
#lm.score(x,y)

In [41]:
print('The R-squared is:',lm.score(z, richard['Deaths - newly reported in last 24 hours']))

The R-squared is: 0.9540426943881454


In [42]:
y_predict_multifit = lm.predict(z)
mse=mean_squared_error(richard['Deaths - newly reported in last 24 hours'], y_predict_multifit)
print('Mean Squared Error:',mse)

Mean Squared Error: 3069.6401478194975


In [44]:
from math import sqrt 

In [45]:
print('Root Mean Squared Error:',sqrt(mse))

Root Mean Squared Error: 55.40433329460339


Decision making;
A model with a a high R-squared and low RMSE is the best.

Multiple Linear Regression Model is the best because it has an accuracy (R-Squared) percenatge of approximately 95.48% with a low RSME of approximately 55.40